lib/eventFactory.js

```diff
@@ -505,13 +545,42 @@ async function createStages({ pipeline, groupEventId }) {
         return Promise.resolve();
     }
 
-    const newStages = await convertStages({ pipeline, stages, groupEventId });
+    // list stage names from this pipeline that already exist
+    const existingStages = await stageFactory.list({ params: { pipelineId, name: stageNames } });
+    const existingStageNames = existingStages.map(stage => stage.name);
+    const convertedStages = await convertStages({ pipeline, stages, eventId, jobs, existingStages });
+    const stagesToUpdate = convertedStages.filter(stage => existingStageNames.includes(stage.name));
```

Comment:
We also need to update `archived=true` for stages that previously existed and have been removed now from the latest `screwdriver.yaml` config.